src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -379,11 +375,9 @@ private SearchOptions UpdateSort(SearchOptions searchOptions, Expression searchE
                 // history is always sorted by _lastUpdated.
                 searchOptions = searchOptions.Clone();
 
-                ISearchParameterDefinitionManager searchParameterDefinitionManager = _searchParameterDefinitionManagerResolver.Invoke();
-
                 searchOptions.Sort = new (SearchParameterInfo searchParameterInfo, SortOrder sortOrder)[]
                 {
-                    (searchParameterDefinitionManager.GetSearchParameter(KnownResourceTypes.Resource, SearchParameterNames.LastUpdated), SortOrder.Ascending),
+                    (_fakeLastUpdate, SortOrder.Ascending),
```

Comment:
I am a little concerned that we are passing in a "fake" `SearchParamInfo` object here. We pass these searchOptions to a couple of rewriters. Those should now be updated (current or in the future) in order to handle incomplete `SearchParamInfo` objects. Seems like that could cause potential issues/bugs in the future. Thoughts?

Reply:
It's beyond optimal, but I don't have better ways to break this knot of cycle dependencies. 

If I had infinite time, I would refactor SearchParameterInfo class to have more strict constructors.

Or maybe have different way of handling sort for default parameters.

Considering how many tests we have I think it's safe to make this change.

For a feature, I think it's quite easy to extend this fake parameter since it's located in one place only.